### PR TITLE
Ordonnable collection

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: Backoffice/EventSubscriber/CollectionSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * Class CollectionSubscriber
+ *
+ */
+class CollectionSubscriber implements EventSubscriberInterface
```
Comment (alavieille) [inline Backoffice/EventSubscriber/CollectionSubscriber.php]: can you rename this subsriber by a name more explicit

File: Backoffice/EventSubscriber/CollectionSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     * @param FormEvent $event
+     *
+     * @return FormInterface
+     */
+    public function preSubmit(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $form->getData();
+
+        if(is_array($event->getData()) && $data instanceof Collection) {
```
Comment (alavieille) [inline Backoffice/EventSubscriber/CollectionSubscriber.php]: psr2

File: Backoffice/EventSubscriber/CollectionSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * Class CollectionSubscriber
+ *
```
Comment (alavieille) [inline Backoffice/EventSubscriber/CollectionSubscriber.php]: remove useless line

File: Backoffice/Form/Type/Extension/CollectionExtension.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    /**
+     * @param FormView      $view
+     * @param FormInterface $form
+     * @param array         $options
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        if ($options['sortable']) {
+            $view->vars['attr'] = array_merge_recursive($view->vars['attr'], array('class' => 'collection-sortable'));
```
Comment (alavieille) [inline Backoffice/Form/Type/Extension/CollectionExtension.php]: what happen if collection have already a class ?

File: Backoffice/Tests/EventSubscriber/CollectionSubscriberTest.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    protected $subscriber;
+
+    /**
+     * Set up the test
+     */
+    public function setUp()
+    {
+        $this->subscriber = new CollectionSubscriber();
+
```
Comment (alavieille) [inline Backoffice/Tests/EventSubscriber/CollectionSubscriberTest.php]: remove this line

**Comment by itkg-nanne** [comment]
update done @alavieille

File: Backoffice/EventSubscriber/SortableCollectionSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public function preSubmit(FormEvent $event)
+    {
+        $form = $event->getForm();
+        $data = $form->getData();
+
+        if (is_array($event->getData())) {
+            $order = array_flip(array_keys($event->getData()));
+            if ($data instanceof Collection) {
+
```
Comment (alavieille) [inline Backoffice/EventSubscriber/SortableCollectionSubscriber.php]: remove this useless line

File: Backoffice/EventSubscriber/SortableCollectionSubscriber.php [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                foreach ($order as $key => $value) {
+                    $child = $data->get($key);
+                    $data->remove($key);
+                    $data->set($key, $child);
+                }
+            }
+            if (is_array($data)) {
```
Comment (alavieille) [inline Backoffice/EventSubscriber/SortableCollectionSubscriber.php]: can you add a test to this part

**Comment by itkg-nanne** [comment]
update done @alavieille

File: Backoffice/Form/Type/Extension/CollectionExtension.php [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    /**
+     * @param FormView      $view
+     * @param FormInterface $form
+     * @param array         $options
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        if ($options['sortable']) {
+            if(array_key_exists('class', $view->vars['attr'])) {
```
Comment (alavieille) [inline Backoffice/Form/Type/Extension/CollectionExtension.php]: psr2

**Pull request merged**